backend/src/test/java/com/google/sps/FindMatchQueryTest.java

```diff
@@ -0,0 +1,151 @@
+    assertThat(actual.getDuration() == DURATION_15_MINUTES);
+  }
+
+  @Test
+  public void compatibleTime() {
+    /* Two participants who are compatible in available time but NOT duration */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB));
+
+    assertThat(Objects.isNull(actual));
+  }
+
+  @Test
+  public void compatibleDuration() {
+    /* Two participants who are compatible in duration but NOT available time */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0250PM, TIMEZONE_DEFAULT, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB));
+    
+    assertThat(Objects.isNull(actual));
+  }
+
+  @Test
+  public void threeParticipants13() {
+    /* Three participants, 1st and 2nd aren't compatible, but 1st and 3rd are */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0250PM, TIMEZONE_DEFAULT, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantC = new Participant(ID_DEFAULT, PERSON_C, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB, participantC));
+    
+    assertThat(actual.getFirstParticipant().getLdap().equals(PERSON_C));
+    assertThat(actual.getSecondParticipant().getLdap().equals(PERSON_A));
+    assertThat(actual.getDuration() == DURATION_60_MINUTES);
+  }
+
+  @Test
+  public void threeParticipants23() {
+    /* Three participants, 1st and 2nd aren't compatible, but 2nd and 3rd are */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0250PM, TIMEZONE_DEFAULT, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantC = new Participant(ID_DEFAULT, PERSON_C, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB, participantC));
+    
+    assertThat(actual.getFirstParticipant().getLdap().equals(PERSON_C));
+    assertThat(actual.getSecondParticipant().getLdap().equals(PERSON_A));
+    assertThat(actual.getDuration() == DURATION_60_MINUTES);
+  }
+
+  @Test
```

Comment:
What do you think about adding a test where there are three participants (A,B,C) where both A-C and B-C match, but the algorithm only returns back one of these matches?

Reply:
Thanks for the suggestion, added! threeParticipantsTwoMatches().